packages/loopring_v3/contracts/impl/libtransactions/TransferTransaction.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2017 Loopring Technology Limited.
+pragma solidity ^0.6.10;
+pragma experimental ABIEncoderV2;
+
+import "../../iface/ExchangeData.sol";
+import "../../thirdparty/BytesUtil.sol";
+import "../../lib/EIP712.sol";
+import "../../lib/FloatUtil.sol";
+import "../../lib/MathUint.sol";
+import "../../lib/SignatureUtil.sol";
+
+
+/// @title TransferTransaction
+/// @author Brecht Devos - <brecht@loopring.org>
+library TransferTransaction
+{
+    using BytesUtil            for bytes;
+    using FloatUtil            for uint;
+    using MathUint             for uint;
+    using SignatureUtil        for bytes32;
+
+    bytes32 constant public TRANSFER_TYPEHASH = keccak256(
+        "Transfer(address from,address to,uint16 tokenID,uint256 amount,uint16 feeTokenID,uint256 fee,uint256 data,uint32 nonce)"
```

Comment:
Do we want to add a validUitll as in https://github.com/Loopring/protocols/issues/1209

Reply:
`validUntil` is available for transfers, but only usable with EdDSA signatures. The reason is that we would need to add the timestamp to the DA data so we have it available onchain. If you think it's important to have even in the case an EdDSA signature isn't used for the transfer it can easily be added.